transformations/writing_system_replacement/transformation.py

```diff
@@ -0,0 +1,502 @@
+"""
+The code is based on the code from the whitespace_perturbation task, created by Xinyi Wu:
+https://github.com/GEM-benchmark/NL-Augmenter/tree/main/transformations/whitespace_perturbation
+"""
+import string
+import numpy as np
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+"""
+"""
+
+replace_white_spaces7 = False # if True, spaces will be considered as regular characters
+deletion_tag = "<del>" # how removed characters are marked in replacement reports
+
+
+def build_chinese_chars():
+    """Returns a string containing many thousands of Chinese characters.
+
+    We use the characters to generate novel alphabets and other writing systems.
+
+    >>> res0 = build_chinese_chars()
+    >>> len(res0)
+    27540
+    >>> res0[:60]
+    '一丁丂七丄丅丆万丈三上下丌不与丏丐丑丒专且丕世丗丘丙业丛东丝丞丟丠両丢丣两严並丧丨丩个丫丬中丮丯丰丱串丳临丵丶丷丸丹为主'
+    >>> res0[-60:]
+    '䵹䵺䵻䵼䵽䵾䵿䶀䶁䶂䶃䶄䶅䶆䶇䶈䶉䶊䶋䶌䶍䶎䶏䶐䶑䶒䶓䶔䶕䶖䶗䶘䶙䶚䶛䶜䶝䶞䶟䶠䶡䶢䶣䶤䶥䶦䶧䶨䶩䶪䶫䶬䶭䶮䶯䶰䶱䶲䶳䶴'
+    """
+    cjk_codes = """
+    4E00–9FDF
+    3400–4DB5
+    """
+    """You use the ranges below too, to greatly increase the 
+    number of available characters.
+    But many systems can't render them correctly yet (as of 24 jul 2021).
+    Thus, enable them only if you're sure that characters are supported
```

Comment:
wouldn't it be interesting to control that with a boolean argument.

Reply:
Good idea! Added the argument